scanner/src/main/kotlin/ScanController.kt

```diff
@@ -249,7 +250,7 @@ class ScanController(
             scanners.map { scanner ->
                 ScanResult(
                     provenance = UnknownProvenance,
-                    scanner = scanner.details,
+                    scanner = ScannerDetails(scanner.name, scanner.version, scanner.configuration),
```

Comment:
That seems very inconvenient to repeat the creation of `ScannerDetails` in so many places. Can you add either a lazy `details` property to `ScannerWrapper` or a `getDetails()` function?

Reply:
The point was exactly to get rid of a `ScannerDetails` property / function in any of the base classes / interfaces of `CommandLinePathScannerWrapper` to avoid the potential misuse. I don't really see a better and safe way other than repeating this. But I'm absolutely open to suggestions.

Reply:
> it's easy to avoid calling it there.

Well, at least I fell into this pitfall about 3 times over the last years...

> It makes sense to me to have name, version and configuration as separate properties, but not having a convenient way to create the details object does not.

Or should we still try to pursue the original idea of this PR to keep `ScannerDetails` in `ScannerWrapper` (and remove `name`from there), plus find a way to forbid use of `ScannerDetails` in `CommandLinePathScannerWrapper`, like by making it private in the base class?

Reply:
But I thought the bootstrapping code needs access to the name so it cannot be part of `ScannerDetails`?

Reply:
> But I thought the bootstrapping code needs access to the name so it cannot be part of `ScannerDetails`?

Yes, sorry, my proposal didn't make sense.

> Also, didn't we plan to remove bootstrapping anyway?

We discussed this at some point, yes, but [in the context of installing ScanCode from binaries](https://github.com/oss-review-toolkit/ort/pull/6105#discussion_r1030833446) I was thinking about using ORT's existing bootstrapping mechanism to also install scanners as part of a Docker build. That's something I'm playing with right now.

Reply:
Ok, I'm giving up on installing ScanCode from binaries as there are issues:

* https://github.com/nexB/scancode-toolkit/issues/2836
* https://github.com/nexB/scancode-toolkit/issues/2731
* https://github.com/nexB/scancode-toolkit/issues/2730
* https://github.com/nexB/scancode-toolkit/issues/769

Looks like indeed simply dropping the scanner bootstrapping feature is the way to go.